src/server/routes/system/queue.ts

```diff
@@ -37,11 +49,45 @@ export async function queueStatus(fastify: FastifyInstance) {
     handler: async (req, res) => {
       const { walletAddress } = req.query;
 
+      // Get # queued and sent transactions.
       const { queued, pending } = await getQueueStatus({ walletAddress });
+
+      // Get sent/queued latency of last 1k txs.
+      const recentTransactions = await prisma.transactions.findMany({
+        orderBy: {
+          queuedAt: "desc",
+        },
+        take: 1_000,
```

Comment:
hmm. can we do the percentile (p90,p50) on SQL?

Reply:
We can but I'm not sure if it's more or less performant than doing it memory. Especially when I want to do 4 different ones. Also this code feels a lot more maintainable/readable.